contracts/actions/flashloan/FLEuler.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+pragma experimental ABIEncoderV2;
+
+import "../ActionBase.sol";
+import "../../utils/ReentrancyGuard.sol";
+import "../../interfaces/flashloan/IERC3156FlashLender.sol";
+
+import "../../interfaces/flashloan/IFlashLoanBase.sol";
+import "../../core/strategy/StrategyModel.sol";
+import "../../interfaces/IDSProxy.sol";
+import "../../interfaces/IFLParamGetter.sol";
+
+import "../../interfaces/euler/IEulerMarkets.sol";
+import "../../interfaces/euler/IDToken.sol";
+
+import "../../utils/TokenUtils.sol";
+import "../../utils/SafeMath.sol";
+
+import "./helpers/FLHelper.sol";
+
+contract FLEuler is ActionBase, ReentrancyGuard, IFlashLoanBase, StrategyModel, FLHelper {
+    using TokenUtils for address;
+    using SafeMath for uint256;
+
+    struct EulerPassingData {
```

Comment:
Could we use `FlashLoanParams` struct as thats what every other FLAction is using? `balanceBefore` you can get after token send to proxy.